src/server/start_service.py

```diff
@@ -107,13 +109,17 @@ def __init__(
         self._advertised_port = advertised_port
         self._reconnect_tasks: List[asyncio.Task] = []
 
-    async def start(self, **kwargs):
+    async def start(self, **kwargs) -> None:
         # we include `kwargs` as a hack for the wallet, which for some
         # reason allows parameters to `_start`. This is serious BRAIN DAMAGE,
         # and should be fixed at some point.
         # TODO: move those parameters to `__init__`
         if self._did_start:
             return
+
+        assert self.self_hostname is not None
```

Comment:
Not 100% sure if that assertions are going to generate problems, my expectation is the `self_hostname` and the `daemon_port` is stablished before you try to start the service.

Reply:
:+1: